### ISSUE TITLE
Fully specify image sources

### ISSUE THREAD

**Issue by garyedwards** [kubernetes-sigs/kind#2186]
**What happened**:

```sh
$ kind create cluster
enabling experimental podman provider
Creating cluster "kind" ...
 ✗ Ensuring node image (kindest/node:v1.20.2) 🖼
ERROR: failed to create cluster: failed to pull image "kindest/node@sha256:8f7ea6e7642c0da54f04a7ee10431549c0257315b3a634f6ef2fecaaedb19bab": command "podman pull kindest/node@sha256:8f7ea6e7642c0da54f04a7ee10431549c0257315b3a634f6ef2fecaaedb19bab" failed with error: exit status 125
Command Output: Error: error getting default registries to try: short-name resolution enforced but cannot prompt without a TTY
```

**What you expected to happen**:

```sh
kind create cluster
enabling experimental podman provider
Creating cluster "kind" ...
 ✓ Ensuring node image (kindest/node:v1.20.2) 🖼
 ✓ Preparing nodes 📦
 ✓ Writing configuration 📜
 ✓ Starting control-plane 🕹️
 ✓ Installing CNI 🔌
 ✓ Installing StorageClass 💾
Set kubectl context to "kind-kind"
You can now use your cluster with:

kubectl cluster-info --context kind-kind

Have a nice day! 👋
```

**How to reproduce it (as minimally and precisely as possible)**:

Use a distro with >=Podman 3.0 with `short-name-mode="enforcing"` enabled in `/etc/containers/registries.conf`

Install `kind` as per the documentation.  Try to create a cluster.

**Anything else we need to know?**:

[Container short name mode](https://www.redhat.com/sysadmin/container-image-short-names) for Podman has been set to `enforcing` by default for Fedora 34 and newer.  This brakes the current scripts.

To fix this you can set the setting to `short-name-mode="permissive"`.  This should be added to the [known issues page](https://kind.sigs.k8s.io/docs/user/known-issues).

As a more robust fix, `kind` should fully specify image sources in its scripts.

**Environment:**

- kind version: (use `kind version`): `kind v0.10.0 go1.15.7 linux/amd64`
- Kubernetes version: (use `kubectl version`): N/A
- Docker version: (use `docker info`): Podman `Version: 3.1.0`
- OS (e.g. from `/etc/os-release`): Fedora IoT 34 Beta

**Comment by BenTheElder** [comment]
@aojea podman broke compat with docker again 🤷‍♂️

**Comment by aojea** [comment]
@vrothberg any suggestion or advice here?

**Comment by vrothberg** [comment]
> @vrothberg any suggestion or advice here?

Thanks for pulling me in.

I think there is a number of ways to address it:
1. Change `short-name-mode="permissive"` as suggested above.
2. Use fully-qualified image names (i.e., docker.io/kindest/node).
3. Add the kind images to https://github.com/containers/shortnames.

The blog linked below goes into the details of short-name aliasing and the reasoning behind it.  I would do a mix of 1) and 3) in this particular case.

> @aojea podman broke compat with docker again man_shrugging

Short-name resolution of Docker (always docker.io) and Podman (configurable to any registry) has always been different.  As mentioned above, Podman 3.0 extended that a bit for security reasons. More on that in the blog: https://www.redhat.com/sysadmin/container-image-short-names

**Comment by aojea** [comment]
> Short-name resolution of Docker (always docker.io)

fair enough, this has to break at some point once the ecosystem grew.

@BenTheElder do I remember some PR with you changing to `fully-qualified image` ?

**Comment by BenTheElder** [comment]
It breaks drop in compatibly, I still consider it a break even if we will handle it and even if I dislike how docker handles this special case. It is not configurable in the rest of the ecosystem and lots of code out there will fully resolve short names non configurably.

Using the existing host eases mirror configuration on the host for some of our common user profiles.

We can do our own short name resolution in the podman compatibly layer.

**Comment by BenTheElder** [comment]
To be clearer: we *expect* that users will override this host and use their own repo, in which case they should also mirror kind images they wish to use, or use --image / image field in config to specify a different image.

If we fully specify this across all hosts then we are passing along this breaking change to users of our stable (docker) driver.

podman / fedora may have reasons for these changes, but I highlight this to @aojea as I'm hesitant to ever declare support stable as we regularly receive user complaints that something new is broken. These are _very_ rare on the stable backend.

**Comment by tao12345666333** [comment]
I submited a PR #2230  PTAL.

**Comment by aojea** [comment]
> * Change `short-name-mode="permissive"` as suggested above.

@vrothberg is there a way to set permissive using podman flags or environment variables?

**Comment by vrothberg** [comment]
@aojea, there is no way to do that for `short-name-mode` specifically. But there's `REGISTRIES_CONF_PATH=$path` for `registries.conf`.

**Comment by vrothberg** [comment]
Very good point, @afbjorklund!

The behavior will change slightly with Podman v3.2. If you have `docker.io/library/busybox:latest` in the local storage and you save `busybox`, it will be saved as `docker.io/library/busybox:latest`.

**Comment by afbjorklund** [comment]
We are using `crane` (go-containerregistry) to save, so it doesn't go to the local daemon or to the local podman first...

But we were "lucky" to use gcr.io as our primary, so we haven't seen the same kind of failure from implicit docker.io.

**Comment by aojea** [comment]
this was fixed by https://github.com/kubernetes-sigs/kind/pull/2285
/close

**Comment by biancarosa** [comment]
Was this fixed? I'm getting the same error on v0.11.1 and fedora 35.

**Comment by aojea** [comment]
> Was this fixed? I'm getting the same error on v0.11.1 and fedora 35.

There is no new release since 0.11.1, so you have to install  kind from @main https://github.com/kubernetes-sigs/kind/pull/2285/commits/9ec6695660e203dc220581b66b62247b32e42018
